### ISSUE TITLE
[semi] Semicolon required even after adding one

### ISSUE THREAD

**Issue by KarboniteKream** [typescript-eslint/typescript-eslint#1695]
When setting up ESLint with `semi` rule set to `always`, it complains that there should be a semicolon after `export default interface ... { ... }`. Even after adding a semicolon with `--fix`, ESLint will report the same issue and retry the operation 9 more times.

A reproducible example can be found at https://github.com/KarboniteKream/interface-semi. I produced the example with the latest version of `vue-cli` and removed anything Vue-related. A similar issue was also reported on https://stackoverflow.com/questions/59790657/eslint-says-that-semicolons-are-missing-even-after-adding-them.

**Repro**

```javascript
module.exports = {
  root: true,
  env: {
    node: true
  },
  extends: [
    'plugin:vue/essential',
    '@vue/typescript/recommended'
  ],
  parserOptions: {
    ecmaVersion: 2020
  },
  rules: {
    semi: ['error', 'always']
  }
};
```

```TS
export default interface Interface {
  name: string;
}
```

**Expected Result**
AFAIK the language grammar doesn't require a semicolon after `export default`. However, if it's required, ESLint should only add one semicolon.

**Actual Result**
ESLint will add semicolons until it stops retrying (10 times) since it will always report a missing semicolon.

**Additional Info**
```
  eslint:cli CLI args: [ 'interface.ts', '--fix', '--debug' ] +0ms
  eslint:cli Running on files +4ms
  eslint:config-array-factory Loading JSON config file: /Users/klemen/interface-semi/package.json +0ms
  eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/node_modules/*', '/bower_components/*' ], basePath: '/Users/klemen/interface-semi', loose: false } ] +0ms
  eslint:ignore-pattern   processed: { basePath: '/Users/klemen/interface-semi', patterns: [ '/node_modules/*', '/bower_components/*' ] } +4ms
  eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/node_modules/*', '/bower_components/*' ], basePath: '/Users/klemen/interface-semi', loose: false } ] +0ms
  eslint:ignore-pattern   processed: { basePath: '/Users/klemen/interface-semi', patterns: [ '/node_modules/*', '/bower_components/*' ] } +0ms
  eslint:file-enumerator Start to iterate files: [ 'interface.ts' ] +0ms
  eslint:file-enumerator File: /Users/klemen/interface-semi/interface.ts +1ms
  eslint:cascading-config-array-factory Load config files for /Users/klemen/interface-semi. +0ms
  eslint:cascading-config-array-factory No cache found: /Users/klemen/interface-semi. +1ms
  eslint:config-array-factory Loading JS config file: /Users/klemen/interface-semi/.eslintrc.js +7ms
  eslint:config-array-factory Config file found: /Users/klemen/interface-semi/.eslintrc.js +1ms
  eslint:config-array-factory Loading {extends:"plugin:vue/essential"} relative to /Users/klemen/interface-semi/.eslintrc.js +1ms
  eslint:config-array-factory Loading plugin "vue" from /Users/klemen/interface-semi/.eslintrc.js +0ms
  eslint:config-array-factory Loaded: eslint-plugin-vue@6.2.1 (/Users/klemen/interface-semi/node_modules/eslint-plugin-vue/lib/index.js) +1ms
  eslint:config-array-factory Plugin /Users/klemen/interface-semi/node_modules/eslint-plugin-vue/lib/index.js loaded in: 142ms +142ms
  eslint:config-array-factory Loading {extends:"/Users/klemen/interface-semi/node_modules/eslint-plugin-vue/lib/configs/base.js"} relative to /Users/klemen/interface-semi/node_modules/eslint-plugin-vue/lib/index.js +1ms
  eslint:config-array-factory package.json was not found: Cannot find module '/Users/klemen/interface-semi/node_modules/eslint-plugin-vue/lib/configs/base.js/package.json'
Require stack:
- /Users/klemen/interface-semi/node_modules/eslint-plugin-vue/lib/index.js +0ms
  eslint:config-array-factory Loaded: /Users/klemen/interface-semi/node_modules/eslint-plugin-vue/lib/configs/base.js (/Users/klemen/interface-semi/node_modules/eslint-plugin-vue/lib/configs/base.js) +0ms
  eslint:config-array-factory Loading JS config file: /Users/klemen/interface-semi/node_modules/eslint-plugin-vue/lib/configs/base.js +0ms
  eslint:config-array-factory Loading parser "/Users/klemen/interface-semi/node_modules/vue-eslint-parser/index.js" from /Users/klemen/interface-semi/node_modules/eslint-plugin-vue/lib/configs/base.js +2ms
  eslint:config-array-factory package.json was not found: Cannot find module '/Users/klemen/interface-semi/node_modules/vue-eslint-parser/index.js/package.json'
Require stack:
- /Users/klemen/interface-semi/node_modules/eslint-plugin-vue/lib/configs/base.js +0ms
  eslint:config-array-factory Loaded: /Users/klemen/interface-semi/node_modules/vue-eslint-parser/index.js (/Users/klemen/interface-semi/node_modules/vue-eslint-parser/index.js) +0ms
  eslint:config-array-factory Loading plugin "vue" from /Users/klemen/interface-semi/node_modules/eslint-plugin-vue/lib/configs/base.js +1ms
  eslint:config-array-factory Loaded: eslint-plugin-vue@6.2.1 (/Users/klemen/interface-semi/node_modules/eslint-plugin-vue/lib/index.js) +0ms
  eslint:config-array-factory Plugin /Users/klemen/interface-semi/node_modules/eslint-plugin-vue/lib/index.js loaded in: 0ms +0ms
  eslint:config-array-factory Loading {extends:"@vue/typescript/recommended"} relative to /Users/klemen/interface-semi/.eslintrc.js +2ms
  eslint:config-array-factory package.json was not found: Cannot find module '@vue/eslint-config-typescript/recommended/package.json'
Require stack:
- /Users/klemen/interface-semi/.eslintrc.js +1ms
  eslint:config-array-factory Loaded: @vue/eslint-config-typescript/recommended (/Users/klemen/interface-semi/node_modules/@vue/eslint-config-typescript/recommended.js) +0ms
  eslint:config-array-factory Loading JS config file: /Users/klemen/interface-semi/node_modules/@vue/eslint-config-typescript/recommended.js +0ms
  eslint:config-array-factory Loading {extends:"./index.js"} relative to /Users/klemen/interface-semi/node_modules/@vue/eslint-config-typescript/recommended.js +2ms
  eslint:config-array-factory package.json was not found: Cannot find module './index.js/package.json'
Require stack:
- /Users/klemen/interface-semi/node_modules/@vue/eslint-config-typescript/recommended.js +0ms
  eslint:config-array-factory Loaded: ./index.js (/Users/klemen/interface-semi/node_modules/@vue/eslint-config-typescript/index.js) +1ms
  eslint:config-array-factory Loading JS config file: /Users/klemen/interface-semi/node_modules/@vue/eslint-config-typescript/index.js +0ms
  eslint:config-array-factory Loading {extends:"plugin:@typescript-eslint/eslint-recommended"} relative to /Users/klemen/interface-semi/node_modules/@vue/eslint-config-typescript/index.js +1ms
  eslint:config-array-factory Loading plugin "@typescript-eslint" from /Users/klemen/interface-semi/node_modules/@vue/eslint-config-typescript/index.js +0ms
  eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@2.22.0 (/Users/klemen/interface-semi/node_modules/@typescript-eslint/eslint-plugin/dist/index.js) +4ms
  eslint:config-array-factory Plugin /Users/klemen/interface-semi/node_modules/@typescript-eslint/eslint-plugin/dist/index.js loaded in: 309ms +309ms
  eslint:config-array-factory Loading plugin "@typescript-eslint" from /Users/klemen/interface-semi/node_modules/@vue/eslint-config-typescript/index.js +1ms
  eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@2.22.0 (/Users/klemen/interface-semi/node_modules/@typescript-eslint/eslint-plugin/dist/index.js) +0ms
  eslint:config-array-factory Plugin /Users/klemen/interface-semi/node_modules/@typescript-eslint/eslint-plugin/dist/index.js loaded in: 0ms +0ms
  eslint:config-array-factory Loading {extends:"plugin:@typescript-eslint/recommended"} relative to /Users/klemen/interface-semi/node_modules/@vue/eslint-config-typescript/recommended.js +0ms
  eslint:config-array-factory Loading plugin "@typescript-eslint" from /Users/klemen/interface-semi/node_modules/@vue/eslint-config-typescript/recommended.js +0ms
  eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@2.22.0 (/Users/klemen/interface-semi/node_modules/@typescript-eslint/eslint-plugin/dist/index.js) +1ms
  eslint:config-array-factory Plugin /Users/klemen/interface-semi/node_modules/@typescript-eslint/eslint-plugin/dist/index.js loaded in: 0ms +0ms
  eslint:config-array-factory Loading {extends:"./configs/base.json"} relative to /Users/klemen/interface-semi/node_modules/@typescript-eslint/eslint-plugin/dist/index.js +0ms
  eslint:config-array-factory package.json was not found: Cannot find module './configs/base.json/package.json'
Require stack:
- /Users/klemen/interface-semi/node_modules/@typescript-eslint/eslint-plugin/dist/index.js +0ms
  eslint:config-array-factory Loaded: ./configs/base.json (/Users/klemen/interface-semi/node_modules/@typescript-eslint/eslint-plugin/dist/configs/base.json) +0ms
  eslint:config-array-factory Loading JSON config file: /Users/klemen/interface-semi/node_modules/@typescript-eslint/eslint-plugin/dist/configs/base.json +0ms
  eslint:config-array-factory Loading parser "@typescript-eslint/parser" from /Users/klemen/interface-semi/node_modules/@typescript-eslint/eslint-plugin/dist/configs/base.json +0ms
  eslint:config-array-factory Loaded: @typescript-eslint/parser@2.22.0 (/Users/klemen/interface-semi/node_modules/@typescript-eslint/parser/dist/parser.js) +1ms
  eslint:config-array-factory Loading plugin "@typescript-eslint" from /Users/klemen/interface-semi/node_modules/@typescript-eslint/eslint-plugin/dist/configs/base.json +45ms
  eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@2.22.0 (/Users/klemen/interface-semi/node_modules/@typescript-eslint/eslint-plugin/dist/index.js) +1ms
  eslint:config-array-factory Plugin /Users/klemen/interface-semi/node_modules/@typescript-eslint/eslint-plugin/dist/index.js loaded in: 0ms +0ms
  eslint:config-array-factory Loading parser "/Users/klemen/interface-semi/node_modules/vue-eslint-parser/index.js" from /Users/klemen/interface-semi/node_modules/@vue/eslint-config-typescript/recommended.js +0ms
  eslint:config-array-factory package.json was not found: Cannot find module '/Users/klemen/interface-semi/node_modules/vue-eslint-parser/index.js/package.json'
Require stack:
- /Users/klemen/interface-semi/node_modules/@vue/eslint-config-typescript/recommended.js +0ms
  eslint:config-array-factory Loaded: /Users/klemen/interface-semi/node_modules/vue-eslint-parser/index.js (/Users/klemen/interface-semi/node_modules/vue-eslint-parser/index.js) +0ms
  eslint:cascading-config-array-factory Stop traversing because of 'root:true'. +518ms
  eslint:rules Loading rule 'getter-return' (remaining=274) +0ms
  eslint:rules Loading rule 'no-dupe-args' (remaining=273) +1ms
  eslint:rules Loading rule 'no-dupe-keys' (remaining=272) +1ms
  eslint:rules Loading rule 'no-unreachable' (remaining=271) +1ms
  eslint:rules Loading rule 'valid-typeof' (remaining=270) +0ms
  eslint:rules Loading rule 'no-const-assign' (remaining=269) +1ms
  eslint:rules Loading rule 'no-new-symbol' (remaining=268) +1ms
  eslint:rules Loading rule 'no-this-before-super' (remaining=267) +1ms
  eslint:rules Loading rule 'no-undef' (remaining=266) +0ms
  eslint:rules Loading rule 'no-dupe-class-members' (remaining=265) +1ms
  eslint:rules Loading rule 'no-redeclare' (remaining=264) +0ms
  eslint:rules Loading rule 'no-unused-vars' (remaining=263) +1ms
  eslint:rules Loading rule 'camelcase' (remaining=262) +2ms
  eslint:rules Loading rule 'no-array-constructor' (remaining=261) +12ms
  eslint:rules Loading rule 'no-empty-function' (remaining=260) +1ms
  eslint:rules Loading rule 'no-use-before-define' (remaining=259) +8ms
  eslint:rules Loading rule 'no-var' (remaining=258) +7ms
  eslint:rules Loading rule 'prefer-const' (remaining=257) +1ms
  eslint:rules Loading rule 'prefer-rest-params' (remaining=256) +2ms
  eslint:rules Loading rule 'prefer-spread' (remaining=255) +1ms
  eslint:rules Loading rule 'semi' (remaining=254) +1ms
  eslint:cascading-config-array-factory Configuration was determined: ConfigArray(13) [ { name: 'DefaultIgnorePattern', filePath: '', criteria: null, env: undefined, globals: undefined, ignorePattern: IgnorePattern { patterns: [Array], basePath: '/Users/klemen/interface-semi', loose: false }, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: undefined, settings: undefined }, { name: '.eslintrc.js » plugin:vue/essential » /Users/klemen/interface-semi/node_modules/eslint-plugin-vue/lib/configs/base.js#processors["vue/.vue"]', filePath: '/Users/klemen/interface-semi/node_modules/eslint-plugin-vue/lib/configs/base.js', criteria: { includes: [Array], excludes: null, basePath: '/Users/klemen/interface-semi' }, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: 'vue/.vue', reportUnusedDisableDirectives: undefined, root: undefined, rules: undefined, settings: undefined }, { name: '.eslintrc.js » plugin:vue/essential » /Users/klemen/interface-semi/node_modules/eslint-plugin-vue/lib/configs/base.js', filePath: '/Users/klemen/interface-semi/node_modules/eslint-plugin-vue/lib/configs/base.js', criteria: null, env: { browser: true, es6: true }, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: { error: null, filePath: '/Users/klemen/interface-semi/node_modules/vue-eslint-parser/index.js', id: '/Users/klemen/interface-semi/node_modules/vue-eslint-parser/index.js', importerName: '.eslintrc.js » plugin:vue/essential » /Users/klemen/interface-semi/node_modules/eslint-plugin-vue/lib/configs/base.js', importerPath: '/Users/klemen/interface-semi/node_modules/eslint-plugin-vue/lib/configs/base.js' }, parserOptions: { ecmaVersion: 2018, sourceType: 'module', ecmaFeatures: [Object] }, plugins: { vue: [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'vue/comment-directive': 'error', 'vue/jsx-uses-vars': 'error' }, settings: undefined }, { name: '.eslintrc.js » plugin:vue/essential', filePath: '/Users/klemen/interface-semi/node_modules/eslint-plugin-vue/lib/index.js', criteria: null, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'vue/no-async-in-computed-properties': 'error', 'vue/no-dupe-keys': 'error', 'vue/no-duplicate-attributes': 'error', 'vue/no-parsing-error': 'error', 'vue/no-reserved-keys': 'error', 'vue/no-shared-component-data': 'error', 'vue/no-side-effects-in-computed-properties': 'error', 'vue/no-template-key': 'error', 'vue/no-textarea-mustache': 'error', 'vue/no-unused-components': 'error', 'vue/no-unused-vars': 'error', 'vue/no-use-v-if-with-v-for': 'error', 'vue/require-component-is': 'error', 'vue/require-prop-type-constructor': 'error', 'vue/require-render-return': 'error', 'vue/require-v-for-key': 'error', 'vue/require-valid-default-prop': 'error', 'vue/return-in-computed-property': 'error', 'vue/use-v-on-exact': 'error', 'vue/valid-template-root': 'error', 'vue/valid-v-bind': 'error', 'vue/valid-v-cloak': 'error', 'vue/valid-v-else-if': 'error', 'vue/valid-v-else': 'error', 'vue/valid-v-for': 'error', 'vue/valid-v-html': 'error', 'vue/valid-v-if': 'error', 'vue/valid-v-model': 'error', 'vue/valid-v-on': 'error', 'vue/valid-v-once': 'error', 'vue/valid-v-pre': 'error', 'vue/valid-v-show': 'error', 'vue/valid-v-text': 'error' }, settings: undefined }, { name: '.eslintrc.js » @vue/eslint-config-typescript/recommended » ./index.js » plugin:@typescript-eslint/eslint-recommended', filePath: '/Users/klemen/interface-semi/node_modules/@typescript-eslint/eslint-plugin/dist/index.js', criteria: null, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: undefined, settings: undefined }, { name: '.eslintrc.js » @vue/eslint-config-typescript/recommended » ./index.js » plugin:@typescript-eslint/eslint-recommended#overrides[0]', filePath: '/Users/klemen/interface-semi/node_modules/@typescript-eslint/eslint-plugin/dist/index.js', criteria: { includes: [Array], excludes: null, basePath: '/Users/klemen/interface-semi' }, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'getter-return': 'off', 'no-dupe-args': 'off', 'no-dupe-keys': 'off', 'no-unreachable': 'off', 'valid-typeof': 'off', 'no-const-assign': 'off', 'no-new-symbol': 'off', 'no-this-before-super': 'off', 'no-undef': 'off', 'no-dupe-class-members': 'off', 'no-redeclare': 'off' }, settings: undefined }, { name: '.eslintrc.js » @vue/eslint-config-typescript/recommended » ./index.js', filePath: '/Users/klemen/interface-semi/node_modules/@vue/eslint-config-typescript/index.js', criteria: null, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: { parser: '/Users/klemen/interface-semi/node_modules/@typescript-eslint/parser/dist/parser.js', extraFileExtensions: [Array], ecmaFeatures: [Object] }, plugins: { '@typescript-eslint': [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: undefined, settings: undefined }, { name: '.eslintrc.js » @vue/eslint-config-typescript/recommended » ./index.js#overrides[0]', filePath: '/Users/klemen/interface-semi/node_modules/@vue/eslint-config-typescript/index.js', criteria: { includes: [Array], excludes: null, basePath: '/Users/klemen/interface-semi' }, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'no-unused-vars': 'off' }, settings: undefined }, { name: '.eslintrc.js » @vue/eslint-config-typescript/recommended » plugin:@typescript-eslint/recommended » ./configs/base.json', filePath: '/Users/klemen/interface-semi/node_modules/@typescript-eslint/eslint-plugin/dist/configs/base.json', criteria: null, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: { error: null, filePath: '/Users/klemen/interface-semi/node_modules/@typescript-eslint/parser/dist/parser.js', id: '@typescript-eslint/parser', importerName: '.eslintrc.js » @vue/eslint-config-typescript/recommended » plugin:@typescript-eslint/recommended » ./configs/base.json', importerPath: '/Users/klemen/interface-semi/node_modules/@typescript-eslint/eslint-plugin/dist/configs/base.json' }, parserOptions: { sourceType: 'module' }, plugins: { '@typescript-eslint': [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: undefined, settings: undefined }, { name: '.eslintrc.js » @vue/eslint-config-typescript/recommended » plugin:@typescript-eslint/recommended', filePath: '/Users/klemen/interface-semi/node_modules/@typescript-eslint/eslint-plugin/dist/index.js', criteria: null, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { '@typescript-eslint/adjacent-overload-signatures': 'error', '@typescript-eslint/ban-ts-ignore': 'error', '@typescript-eslint/ban-types': 'error', camelcase: 'off', '@typescript-eslint/camelcase': 'error', '@typescript-eslint/class-name-casing': 'error', '@typescript-eslint/consistent-type-assertions': 'error', '@typescript-eslint/explicit-function-return-type': 'warn', '@typescript-eslint/interface-name-prefix': 'error', '@typescript-eslint/member-delimiter-style': 'error', 'no-array-constructor': 'off', '@typescript-eslint/no-array-constructor': 'error', 'no-empty-function': 'off', '@typescript-eslint/no-empty-function': 'error', '@typescript-eslint/no-empty-interface': 'error', '@typescript-eslint/no-explicit-any': 'warn', '@typescript-eslint/no-inferrable-types': 'error', '@typescript-eslint/no-misused-new': 'error', '@typescript-eslint/no-namespace': 'error', '@typescript-eslint/no-non-null-assertion': 'warn', '@typescript-eslint/no-this-alias': 'error', 'no-unused-vars': 'off', '@typescript-eslint/no-unused-vars': 'warn', 'no-use-before-define': 'off', '@typescript-eslint/no-use-before-define': 'error', '@typescript-eslint/no-var-requires': 'error', '@typescript-eslint/prefer-namespace-keyword': 'error', '@typescript-eslint/triple-slash-reference': 'error', '@typescript-eslint/type-annotation-spacing': 'error', 'no-var': 'error', 'prefer-const': 'error', 'prefer-rest-params': 'error', 'prefer-spread': 'error' }, settings: undefined }, { name: '.eslintrc.js » @vue/eslint-config-typescript/recommended', filePath: '/Users/klemen/interface-semi/node_modules/@vue/eslint-config-typescript/recommended.js', criteria: null, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: { error: null, filePath: '/Users/klemen/interface-semi/node_modules/vue-eslint-parser/index.js', id: '/Users/klemen/interface-semi/node_modules/vue-eslint-parser/index.js', importerName: '.eslintrc.js » @vue/eslint-config-typescript/recommended', importerPath: '/Users/klemen/interface-semi/node_modules/@vue/eslint-config-typescript/recommended.js' }, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { '@typescript-eslint/explicit-function-return-type': 'off' }, settings: undefined }, { name: '.eslintrc.js » @vue/eslint-config-typescript/recommended#overrides[0]', filePath: '/Users/klemen/interface-semi/node_modules/@vue/eslint-config-typescript/recommended.js', criteria: { includes: [Array], excludes: null, basePath: '/Users/klemen/interface-semi' }, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { '@typescript-eslint/no-empty-interface': 'off', '@typescript-eslint/no-explicit-any': 'off', '@typescript-eslint/no-unused-vars': 'off' }, settings: undefined }, { name: '.eslintrc.js', filePath: '/Users/klemen/interface-semi/.eslintrc.js', criteria: null, env: { node: true }, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: { ecmaVersion: 2020 }, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: true, rules: { semi: [Array] }, settings: undefined } ] on /Users/klemen/interface-semi +59ms
  eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/node_modules/*', '/bower_components/*' ], basePath: '/Users/klemen/interface-semi', loose: false } ] +586ms
  eslint:ignore-pattern   processed: { basePath: '/Users/klemen/interface-semi', patterns: [ '/node_modules/*', '/bower_components/*' ] } +3ms
  eslint:ignore-pattern Check {
  filePath: '/Users/klemen/interface-semi/interface.ts',
  dot: false,
  relativePath: 'interface.ts',
  result: false
} +1ms
  eslint:cli-engine Lint /Users/klemen/interface-semi/interface.ts +0ms
  eslint:linter Linting code for /Users/klemen/interface-semi/interface.ts (pass 1) +0ms
  eslint:linter Verify +1ms
  eslint:linter With ConfigArray: /Users/klemen/interface-semi/interface.ts +0ms
  eslint:linter Generating fixed text for /Users/klemen/interface-semi/interface.ts (pass 1) +57ms
  eslint:source-code-fixer Applying fixes +0ms
  eslint:source-code-fixer Found fixes to apply +0ms
  eslint:linter Linting code for /Users/klemen/interface-semi/interface.ts (pass 2) +1ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: /Users/klemen/interface-semi/interface.ts +0ms
  eslint:linter Generating fixed text for /Users/klemen/interface-semi/interface.ts (pass 2) +6ms
  eslint:source-code-fixer Applying fixes +6ms
  eslint:source-code-fixer Found fixes to apply +0ms
  eslint:linter Linting code for /Users/klemen/interface-semi/interface.ts (pass 3) +0ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: /Users/klemen/interface-semi/interface.ts +0ms
  eslint:linter Generating fixed text for /Users/klemen/interface-semi/interface.ts (pass 3) +10ms
  eslint:source-code-fixer Applying fixes +10ms
  eslint:source-code-fixer Found fixes to apply +0ms
  eslint:linter Linting code for /Users/klemen/interface-semi/interface.ts (pass 4) +0ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: /Users/klemen/interface-semi/interface.ts +0ms
  eslint:linter Generating fixed text for /Users/klemen/interface-semi/interface.ts (pass 4) +6ms
  eslint:source-code-fixer Applying fixes +6ms
  eslint:source-code-fixer Found fixes to apply +0ms
  eslint:linter Linting code for /Users/klemen/interface-semi/interface.ts (pass 5) +0ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: /Users/klemen/interface-semi/interface.ts +0ms
  eslint:linter Generating fixed text for /Users/klemen/interface-semi/interface.ts (pass 5) +9ms
  eslint:source-code-fixer Applying fixes +9ms
  eslint:source-code-fixer Found fixes to apply +0ms
  eslint:linter Linting code for /Users/klemen/interface-semi/interface.ts (pass 6) +0ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: /Users/klemen/interface-semi/interface.ts +0ms
  eslint:linter Generating fixed text for /Users/klemen/interface-semi/interface.ts (pass 6) +4ms
  eslint:source-code-fixer Applying fixes +4ms
  eslint:source-code-fixer Found fixes to apply +1ms
  eslint:linter Linting code for /Users/klemen/interface-semi/interface.ts (pass 7) +1ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: /Users/klemen/interface-semi/interface.ts +0ms
  eslint:linter Generating fixed text for /Users/klemen/interface-semi/interface.ts (pass 7) +6ms
  eslint:source-code-fixer Applying fixes +7ms
  eslint:source-code-fixer Found fixes to apply +0ms
  eslint:linter Linting code for /Users/klemen/interface-semi/interface.ts (pass 8) +1ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: /Users/klemen/interface-semi/interface.ts +0ms
  eslint:linter Generating fixed text for /Users/klemen/interface-semi/interface.ts (pass 8) +4ms
  eslint:source-code-fixer Applying fixes +4ms
  eslint:source-code-fixer Found fixes to apply +0ms
  eslint:linter Linting code for /Users/klemen/interface-semi/interface.ts (pass 9) +0ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: /Users/klemen/interface-semi/interface.ts +0ms
  eslint:linter Generating fixed text for /Users/klemen/interface-semi/interface.ts (pass 9) +4ms
  eslint:source-code-fixer Applying fixes +5ms
  eslint:source-code-fixer Found fixes to apply +0ms
  eslint:linter Linting code for /Users/klemen/interface-semi/interface.ts (pass 10) +1ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: /Users/klemen/interface-semi/interface.ts +0ms
  eslint:linter Generating fixed text for /Users/klemen/interface-semi/interface.ts (pass 10) +5ms
  eslint:source-code-fixer Applying fixes +5ms
  eslint:source-code-fixer Found fixes to apply +0ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: /Users/klemen/interface-semi/interface.ts +0ms
  eslint:file-enumerator Complete iterating files: ["interface.ts"] +709ms
  eslint:cli-engine Linting complete in: 711ms +122ms
  eslint:cli Fix mode enabled - applying fixes +728ms

/Users/klemen/interface-semi/interface.ts
  3:2  error  Missing semicolon  semi

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.22.0` |
| `@typescript-eslint/parser`        | `2.22.0` |
| `TypeScript`                       | `3.7.5` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `13.8.0` |
| `npm`                              | `6.13.7` |

**Comment by KarboniteKream** [comment]
I just found https://github.com/typescript-eslint/typescript-eslint/issues/123, which might be related...

**Comment by bradzacher** [comment]
Thanks for the detailed issue - check out our FAQ - https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md#i-am-using-a-rule-from-eslint-core-and-it-doesnt-work-correctly-with-typescript-code

You'll find we have an extension rule.
https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin#extension-rules

**Comment by KarboniteKream** [comment]
Thank you for the quick reply!
